# Bioproject PRJNA644285

This sample comes from the [NCBI BioProject PRJNA644285](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA644285).

# Title:

Study of fungal clinical isolates from different anatomical sites and metagenomic of the lung microbiome of patients with cystic fibrosis

# Description:

Microorganisms and their balance in the human body are necessary to maintain the health. Once this equilibrium is broken, the excessive growth of pathogens occurs. Several factors lead to this imbalance, one of them is the mutation in the Transmembrane Conductivity Regulator gene of Cystic Fibrosis that causes the interruption of the normal transfer of chlorine and sodium ions through the cell membranes that lead to dehydration and consequently increase of the viscosity of cellular secretions, which may lead to failure of the mucociliary pulmonary clearance of patients with cystic fibrosis (CF). The study of microbiome has been facilitated by the Next Generation Sequencing technology capable of even sequencing genetic material from non-cultivable microorganisms. The present project aims to study by metagenomics, the microbiome (bacteria and fungi) of the lung of patients with cystic fibrosis before and after antibiotic therapy; knowing the fungi and bacteria as well as the inventory of the genes related to virulence, antimicrobial resistance and others. Additionally, clinical isolates of pathogenic filamentous fungi of Aspergillus and Fusarium genera from several anatomical sites, including sputum of CF patients will be studied. The molecular identification, sensitivity to antifungal and antimicrobial photodynamic therapy with phenotiazinium photosensitizers, study of the virulence factors and virulence in the experimental model Galleria mellonella will bring important information regarding these pathogenic microorganisms. The results obtained here will provide a critical view and assist the clinical decision-making regarding the best therapeutic approach.


# Metadata:
We have included the metadata in two separate files, a [JSON format file](PRJNA644285.metadata.json.gz) that is computer readable, or a [tsv format file](PRJNA644285.metadata.tsv.gz) that you can import into excel or similar.

# Publications

de Almeida OGG, Capizzani CP da C, Tonani L, Grizante Bariao PH, da Cunha AF, De Martinis ECP, Torres LAGMM, von Zeska Kress MR. 2020. The lung microbiome of three young Brazilian patients with cystic fibrosis colonized by fungi. [Front Cell Infect Microbiol 10:598938](https://doi.org/10.3389/fcimb.2020.598938).
  
# Analysis:

- The entire data set is 12 runs, and 29,679,938,510 bp
- We analysed 12 metagenomic sequence runs.
- We predicted 3 samples out of 12 (25.0%) have _Pseudomonas aeruginosa_


## Prediction outcomes


Sample | pwCF | Data |  Date  |  Clinical condition |  Fev-01 |  Culture |   Antibiotic Therapy | Pseudomonas Prediction | Confidence | Certainty | Outcome | 
 --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | 
SRR12149839 | PA1 | PQ\_589\_R1 |  7/07/2018 |  Exacerbation | 39% | A. fumigatus, S. aureus MRSA, P. aeruginosa, Burkholderia. cepacia complex | Ciprofloxacin, Tobramycin, Amoxi/Clav | Negative | 0.65 | Medium | Incorrect | 
SRR12149836 | PA2 | PQ\_590\_R1 |  08/24/2018 |  Exacerbation |   |  NI |   | Negative | 0.72 | Medium | Correct | 
SRR12149835 | PA3 | PQ\_591\_R1 |  09/14/2018 |  Ambulatory |   | A. fumigatus, Burkholderia cepacia complex |   | Negative | 0.67 | Medium | Correct | 
SRR12149830 | PA4 | PQ\_596\_R1 |  10/26/2018 |  Exacerbation |   | Aspergillus fumigatus, Penicillium sp. | Meropenem, Vancomicin, Gentamycin | Negative | 0.64 | Medium | Correct | 
SRR12149837 | PA5 | PQ\_599\_R1 |  11/30/2018 |  Ambulatory |   | Aspergillus fumigatus, Torulaspora delbrueckii |  | Negative | 0.71 | Medium | Correct | 
SRR12149834 | PB1 | PQ\_592\_R1 |  08/23/2018 |  Exacerbation | 64% |  NI | Ceftazidime, Gentamicine | Negative | 0.71 | Medium | Correct | 
SRR12149829 | PB2 | PQ\_597\_R1 | 10/26/2018 |  Ambulatory |  | Talaromyces sp., Torulaspora delbrueckii, Hanseniaspora sp., Burkholderia cepacia complex |  | Negative | 0.69 | Medium | Correct | 
SRR12149838 | PB3 | PQ\_598\_R1 |  11/30/2018 |  Ambulatory |   |  Talaromyces amestolkiae |   | Negative | 0.56 | Low | Correct | 
SRR12149840 | PC1 | PQ\_588\_R1 |  05/18/2018 |  Exacerbation | 37% | Aspergillus fumigatus, Candida albicans, Achromobacter sp. | Itraconazole | Positive | 0.59 | Low | Correct | 
SRR12149833 | PC2 | PQ\_593\_R1 |  08/31/2018 |  Exacerbation |   | Aspergillus fumigatus, Pseudmonas aeruginosa, Stenotrophomonas maltophilia | Meropenem, Gentamicin | Positive | 0.59 | Low | Correct | 
SRR12149832 | PC3 | PQ\_594\_R1 | 09/14/2018 | Exacerbation |  | Aspergillus fumigatus, Achromobacter xylosoxidans, Chryseobacterium indologenes |  | Negative | 0.53 | Low | Correct | 
SRR12149831 | PC4 | PQ\_595\_R1 | 10/26/2018 | Ambulatory |  | Aspergillus fumigatus, Pseudomonas  aeruginosa, Staphylococcus aureus |  | Positive | 0.66 | Medium | Correct | 


## t-SNE
![Comparison of Adelaide and PRJNA644285 samples by t-SNE](img/PRJNA644285_Pseudomonas_tSNE.png 'Fig. t-SNE of all the analysed sequence data coloured by whether Pseudomonas is predicted')


## PCA
![This cluster of features are most strongly associated with the presence of Pseudomonas](img/PRJNA644285_Pseudomonas_PCA.png 'Fig. PCA of the cluster of features most strongly associated with Pseudomonas colonization in PRJNA644285')


