# Bioproject PRJEB54014

This sample comes from the [NCBI BioProject PRJEB54014](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJEB54014).

# Title:
Sputum_Metagenomics

# Description:
A pan-genome resembling genome-scale metabolic model platform of 252 Aspergillus fumigatus strains reveals growth dependencies from the lung microbiome

The shotgun metagenomics of sputum from 40 cystic fibrosis patients before and after they were diagnosed with an A. fumigatus infection


# Metadata:
We have included the metadata in two separate files, a [JSON format file](PRJEB54014.metadata.json.gz) that is computer readable, or a [tsv format file](PRJEB54014.metadata.tsv.gz) that you can import into excel or similar.


# Publications

Mirhakkak MH, Chen X, Ni Y, Heinekamp T, Sae-Ong T, Xu L-L, Kurzai O, Barber AE, Brakhage AA, Boutin S, Sch√§uble S, Panagiotou G. 2023. Genome-scale metabolic modeling of Aspergillus fumigatus strains reveals growth dependencies on the lung microbiome. [Nat Commun 14:4369](https://www.nature.com/articles/s41467-023-39982-5).
  
# Analysis:

# BioProject: PRJEB54014

- The entire data set is 80 runs, and 447,488,092,200 bp (including amplicon sequences)
- We analysed 75 metagenomic sequence runs.
- We predicted 21 samples out of 75 (28.0%) have _Pseudomonas aeruginosa_

<details>
<summary>
## Prediction outcomes
</summary>

Sample | Pseudomonas Prediction | Confidence | Certainty
 --- | --- | --- | ---
ERR9901908 | Negative | 0.72 | Medium
ERR9901951 | Positive | 0.75 | Medium
ERR9901955 | Positive | 0.71 | Medium
ERR9901956 | Negative | 0.76 | Medium
ERR9901958 | Negative | 0.69 | Medium
ERR9901960 | Positive | 0.54 | Low
ERR9901961 | Positive | 0.67 | Medium
ERR9901963 | Positive | 0.66 | Medium
ERR9901964 | Negative | 0.82 | High
ERR9901966 | Positive | 0.78 | Medium
ERR9901967 | Negative | 0.61 | Medium
ERR9901968 | Negative | 0.68 | Medium
ERR9901969 | Negative | 0.63 | Medium
ERR9901970 | Negative | 0.74 | Medium
ERR9901972 | Negative | 0.77 | Medium
ERR9901974 | Positive | 0.74 | Medium
ERR9901975 | Positive | 0.72 | Medium
ERR9901976 | Negative | 0.68 | Medium
ERR9901977 | Positive | 0.76 | Medium
ERR9901978 | Positive | 0.76 | Medium
ERR9901979 | Negative | 0.82 | High
ERR9901980 | Negative | 0.82 | High
ERR9901981 | Negative | 0.74 | Medium
ERR9901982 | Negative | 0.90 | High
ERR9901983 | Negative | 0.73 | Medium
ERR9901984 | Positive | 0.69 | Medium
ERR9901985 | Positive | 0.75 | Medium
ERR9901986 | Negative | 0.64 | Medium
ERR9901987 | Positive | 0.77 | Medium
ERR9901988 | Negative | 0.72 | Medium
ERR9901989 | Negative | 0.81 | High
ERR9901990 | Negative | 0.82 | High
ERR9901991 | Negative | 0.71 | Medium
ERR9901992 | Negative | 0.81 | High
ERR9901993 | Negative | 0.63 | Medium
ERR9901994 | Negative | 0.71 | Medium
ERR9901995 | Positive | 0.74 | Medium
ERR9901996 | Negative | 0.83 | High
ERR9901997 | Negative | 0.59 | Low
ERR9901998 | Negative | 0.62 | Medium
ERR9901999 | Positive | 0.77 | Medium
ERR9902000 | Negative | 0.68 | Medium
ERR9902003 | Negative | 0.67 | Medium
ERR9902005 | Negative | 0.71 | Medium
ERR9902018 | Negative | 0.69 | Medium
ERR9902019 | Negative | 0.77 | Medium
ERR9902020 | Negative | 0.75 | Medium
ERR9902021 | Positive | 0.56 | Low
ERR9902023 | Negative | 0.72 | Medium
ERR9902025 | Positive | 0.65 | Medium
ERR9902026 | Negative | 0.50 | Low
ERR9902027 | Positive | 0.72 | Medium
ERR9902028 | Negative | 0.77 | Medium
ERR9902031 | Negative | 0.71 | Medium
ERR9902033 | Negative | 0.58 | Low
ERR9902035 | Positive | 0.75 | Medium
ERR9902036 | Negative | 0.85 | High
ERR9902037 | Negative | 0.58 | Low
ERR9902038 | Negative | 0.71 | Medium
ERR9902039 | Positive | 0.70 | Medium
ERR9902040 | Negative | 0.75 | Medium
ERR9902041 | Negative | 0.71 | Medium
ERR9902043 | Negative | 0.82 | High
ERR9902044 | Negative | 0.62 | Medium
ERR9902046 | Negative | 0.63 | Medium
ERR9902047 | Negative | 0.72 | Medium
ERR9902048 | Negative | 0.76 | Medium
ERR9902049 | Negative | 0.55 | Low
ERR9902050 | Negative | 0.76 | Medium
ERR9902051 | Negative | 0.77 | Medium
ERR9902052 | Negative | 0.64 | Medium
ERR9902053 | Negative | 0.57 | Low
ERR9902054 | Negative | 0.60 | Medium
ERR9902055 | Negative | 0.66 | Medium
ERR9902056 | Positive | 0.72 | Medium
</details>

## t-SNE
![Comparison of Adelaide and PRJEB54014 samples by t-SNE](img/PRJEB54014_Pseudomonas_tSNE.png 'Fig. t-SNE of all the analysed sequence data coloured by whether Pseudomonas is predicted')


## PCA
![This cluster of features are most strongly associated with the presence of Pseudomonas](img/PRJEB54014_Pseudomonas_PCA.png 'Fig. PCA of the cluster of features most strongly associated with Pseudomonas colonization in PRJEB54014')


