{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b8617db5-c2c7-40fb-9cbc-a01cc326e9cb",
   "metadata": {},
   "source": [
    "# PCA Figure\n",
    "\n",
    "A recapitulation of Jess' primer figure but written in python by Rob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a6e5e004-ec06-490c-865e-ee0ae1db9d19",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "\n",
    "import re\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.collections import PatchCollection\n",
    "import matplotlib.colors as mcolors\n",
    "import matplotlib.dates as mdates\n",
    "from matplotlib.colors import ListedColormap\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import json\n",
    "\n",
    "from itertools import cycle\n",
    "\n",
    "from sklearn.decomposition import PCA\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from sklearn.inspection import permutation_importance\n",
    "\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.ensemble import RandomForestRegressor, RandomForestClassifier\n",
    "from sklearn.metrics import mean_squared_error\n",
    "\n",
    "from scipy.stats import linregress\n",
    "\n",
    "\n",
    "# there is a FutureWarning in sklearn StandardScalar which is really annoying. This ignores it.\n",
    "import warnings\n",
    "warnings.simplefilter(action='ignore', category=FutureWarning)\n",
    "\n",
    "try:\n",
    "  import google.colab\n",
    "  IN_COLAB = True\n",
    "  !pip install adjustText\n",
    "  from google.colab import drive\n",
    "  drive.mount('/content/drive')\n",
    "  datadir = '/content/drive/MyDrive/Projects/CF/Adelaide/CF_Data_Analysis'\n",
    "except ImportError:\n",
    "  IN_COLAB = False\n",
    "  datadir = '..'\n",
    "\n",
    "from adjustText import adjust_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8fffc305-eee0-459f-96ff-a31b0a198b8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "from cf_analysis_lib import read_taxonomy, read_metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "41da48ae-3711-4b66-928c-d8807c987a0f",
   "metadata": {},
   "outputs": [],
   "source": [
    "datadir = \"..\"\n",
    "sequence_type = \"MGI\"\n",
    "taxa = \"genus\"\n",
    "df = read_taxonomy(datadir, sequence_type, taxa)\n",
    "df = df.T\n",
    "df.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4914f557-67ab-4253-9d1c-e53d11f23245",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = read_metadata(datadir, sequence_type)\n",
    "metadata.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4d1a5ced-5923-4ec6-b489-ff86eb95e102",
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.kdeplot(data=df['Pseudomonas'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "caeaf1b2-a912-4d90-9b78-2f857b552c5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "genus = 'Pseudomonas'\n",
    "list(df[df[genus] < 100][genus])\n",
    "df[df[genus] > 0].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5f827423-6a93-4049-8107-6397ac9596d3",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Create some bins for the sizes\n",
    "# Define the bins and labels\n",
    "# Non zero mean is the mean of all the non zero values\n",
    "genus = 'Pseudomonas'\n",
    "mn = np.mean(df[df[genus] > 0][genus])\n",
    "md = np.median(df[df[genus] > 0][genus])\n",
    "print(f\"The mean for {genus} is {mn} and the median is {md}\", file=sys.stderr)\n",
    "\n",
    "nonzeromean = np.median(df[df[genus] > 0][genus])\n",
    "bins = [-1, 0, nonzeromean/2, nonzeromean*2, float('inf')]  # Include -1 to capture 0 in 'small'\n",
    "#bins = [-1, 200, 5000, 10000, float('inf')]\n",
    "labels = ['small', 'medium', 'large']\n",
    "sizelist = [50, 200, 500, 1000]\n",
    "marker_shapes = ['s', 'V', '^', 'o'] # see https://seaborn.pydata.org/tutorial/properties.html#marker\n",
    "\n",
    "# Create a new column 'size' based on the conditions\n",
    "sizes = list(pd.cut(df['Pseudomonas'], bins=bins, labels=sizelist))\n",
    "markers = list(pd.cut(df['Pseudomonas'], bins=bins, labels=marker_shapes))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "66f24f97-fc05-419b-98a1-0f47bd0cf95b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Normalization/Transformation\n",
    "# Square root transform\n",
    "sqrt_df = np.sqrt(df)\n",
    "\n",
    "# perform the PCA\n",
    "pca = PCA(n_components=2)\n",
    "\n",
    "pca_result = pca.fit_transform(sqrt_df)\n",
    "# Create a DataFrame with PCA results\n",
    "pca_df = pd.DataFrame(data=pca_result, index=df.index, columns=['PC1', 'PC2'])\n",
    "\n",
    "# Get loadings\n",
    "loadings = pca.components_.T * np.sqrt(pca.explained_variance_)\n",
    "# Create a DataFrame for loadings with feature names\n",
    "# Select the top 5 most important loadings\n",
    "loadings_df = pd.DataFrame(loadings, index=df.columns, columns=['PC1', 'PC2'])\n",
    "\n",
    "# Create a DataFrame for top loadings\n",
    "top_loadings_df = loadings_df.loc[loadings_df['PC1'].abs().sort_values(ascending=False).index]\n",
    "top_loadings_df.head()\n",
    "\n",
    "explained_variance = pca.explained_variance_ratio_ * 100\n",
    "pc1_variance = explained_variance[0]\n",
    "pc2_variance = explained_variance[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ddffccb8-afeb-4030-89aa-afab02646617",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Map colors based on the metadata column\n",
    "colours = np.where(metadata['CS_Pseudomonas aeruginosa'] == 1, 'blue', 'red')\n",
    "\n",
    "# Plot the PCA results\n",
    "plt.figure(figsize=(16, 12))\n",
    "plt.scatter(pca_df['PC1'], pca_df['PC2'], alpha=0.2, c=colours, s=sizes)\n",
    "plt.title('Pseudomonas detected by MGI sequencing coloured whether positive - present' + f\"\\nMedian number of '{genus}' sequences: {md:.0f}\")\n",
    "plt.xlabel(f'Principal Component 1 ({pc1_variance:.3f}%)')\n",
    "plt.ylabel(f'Principal Component 2 ({pc2_variance:.3f}%)')\n",
    "\n",
    "# add the loadings ... we only plot maxloadings here\n",
    "maxloadings = 15\n",
    "if len(loadings) < maxloadings:\n",
    "    maxloadings = len(loadings)\n",
    "\n",
    "plotscaler = 2\n",
    "texts = []\n",
    "colour_cycle = cycle(mcolors.TABLEAU_COLORS)\n",
    "found_pseudomonas = False\n",
    "for i in range(maxloadings):\n",
    "    c = next(colour_cycle)\n",
    "    if top_loadings_df.index[i] == 'Pseudomonas':\n",
    "        found_pseudomonas = True\n",
    "        c = 'black'\n",
    "    xpos = top_loadings_df.iloc[i, 0]*plotscaler\n",
    "    ypos = top_loadings_df.iloc[i, 1]*plotscaler\n",
    "    plt.arrow(0, 0, xpos, ypos, \n",
    "              color=c, alpha=0.5, width=0.05)\n",
    "    texts.append(plt.text(xpos, ypos, top_loadings_df.index[i], color=c))\n",
    "\n",
    "if not found_pseudomonas:\n",
    "    # specifically add Pseudomonas\n",
    "    pidx = top_loadings_df.index.get_loc('Pseudomonas')\n",
    "    xpos = top_loadings_df.iloc[pidx, 0]*plotscaler\n",
    "    ypos = top_loadings_df.iloc[pidx, 1]*plotscaler\n",
    "    plt.arrow(0, 0, xpos, ypos, color='black', alpha=0.5, width=0.05)\n",
    "    texts.append(plt.text(xpos, ypos, top_loadings_df.index[pidx], color='black'))\n",
    "\n",
    "\n",
    "    \n",
    "adjust_text(texts)\n",
    "\n",
    "# Add a legend\n",
    "blue_patch = plt.Line2D([0], [0], marker='o', color='w', label='Pseudomonas culture positive', \n",
    "                         markerfacecolor='blue', alpha=0.2, markersize=10)\n",
    "red_patch = plt.Line2D([0], [0], marker='o', color='w', label='Pseudomonas culture negative', \n",
    "                        markerfacecolor='red', alpha=0.2, markersize=10)\n",
    "small_patch = plt.scatter([], [], color='k', label=f'{bins[1]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[0])\n",
    "med1_patch = plt.scatter([], [], color='k', label=f'{bins[1]:.0f} - {bins[2]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[1])\n",
    "med2_patch = plt.scatter([], [], color='k', label=f'{bins[2]:.0f} - {bins[3]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[2])\n",
    "big_patch = plt.scatter([], [], color='k', label=f'>{bins[3]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[3])\n",
    "\n",
    "plt.legend(handles=[blue_patch, red_patch, small_patch, med1_patch, med2_patch, big_patch])\n",
    "\n",
    "\n",
    "plt.savefig('img/MGI_genus_pseudomonas_present.png')\n",
    "# Show the plot\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f3232497-fe2f-42ba-a4e6-23f50cbacbc0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Map colors based on the metadata column\n",
    "columns_to_check = ['CS_Pseudomonas aeruginosa', 'P12M_2_Pseudomonas aeruginosa']\n",
    "colours = metadata[columns_to_check].isin([1]).any(axis=1).map({True: 'blue', False: 'red'})\n",
    "\n",
    "# Plot the PCA results\n",
    "plt.figure(figsize=(16, 12))\n",
    "plt.scatter(pca_df['PC1'], pca_df['PC2'], alpha=0.2, c=colours, s=sizes)\n",
    "plt.title('Pseudomonas detected by MGI sequencing coloured whether positive - past or present' + f\"\\nMedian number of '{genus}' sequences: {md:.0f}\")\n",
    "plt.xlabel(f'Principal Component 1 ({pc1_variance:.3f}%)')\n",
    "plt.ylabel(f'Principal Component 2 ({pc2_variance:.3f}%)')\n",
    "\n",
    "# add the loadings ... we only plot maxloadings here\n",
    "maxloadings = 15\n",
    "if len(loadings) < maxloadings:\n",
    "    maxloadings = len(loadings)\n",
    "\n",
    "plotscaler = 2\n",
    "texts = []\n",
    "colour_cycle = cycle(mcolors.TABLEAU_COLORS)\n",
    "found_pseudomonas = False\n",
    "for i in range(maxloadings):\n",
    "    c = next(colour_cycle)\n",
    "    if top_loadings_df.index[i] == 'Pseudomonas':\n",
    "        found_pseudomonas = True\n",
    "        c = 'black'\n",
    "    xpos = top_loadings_df.iloc[i, 0]*plotscaler\n",
    "    ypos = top_loadings_df.iloc[i, 1]*plotscaler\n",
    "    plt.arrow(0, 0, xpos, ypos, \n",
    "              color=c, alpha=0.5, width=0.05)\n",
    "    texts.append(plt.text(xpos, ypos, top_loadings_df.index[i], color=c))\n",
    "\n",
    "if not found_pseudomonas:\n",
    "    # specifically add Pseudomonas\n",
    "    pidx = top_loadings_df.index.get_loc('Pseudomonas')\n",
    "    xpos = top_loadings_df.iloc[pidx, 0]*plotscaler\n",
    "    ypos = top_loadings_df.iloc[pidx, 1]*plotscaler\n",
    "    plt.arrow(0, 0, xpos, ypos, color='black', alpha=0.5, width=0.05)\n",
    "    texts.append(plt.text(xpos, ypos, top_loadings_df.index[pidx], color='black'))\n",
    "\n",
    "\n",
    "    \n",
    "adjust_text(texts)\n",
    "\n",
    "# Add a legend\n",
    "blue_patch = plt.Line2D([0], [0], marker='o', color='w', label='Pseudomonas culture positive', \n",
    "                         markerfacecolor='blue', alpha=0.2, markersize=10)\n",
    "red_patch = plt.Line2D([0], [0], marker='o', color='w', label='Pseudomonas culture negative', \n",
    "                        markerfacecolor='red', alpha=0.2, markersize=10)\n",
    "small_patch = plt.scatter([], [], color='k', label=f'{bins[1]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[0])\n",
    "med1_patch = plt.scatter([], [], color='k', label=f'{bins[1]:.0f} - {bins[2]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[1])\n",
    "med2_patch = plt.scatter([], [], color='k', label=f'{bins[2]:.0f} - {bins[3]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[2])\n",
    "big_patch = plt.scatter([], [], color='k', label=f'>{bins[3]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[3])\n",
    "\n",
    "plt.legend(handles=[blue_patch, red_patch, small_patch, med1_patch, med2_patch, big_patch])\n",
    "\n",
    "\n",
    "plt.savefig('img/MGI_genus_pseudomonas_past_present.png')\n",
    "# Show the plot\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "349a4558-41b2-45c0-956d-292d05504347",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Map colors based on the metadata column\n",
    "columns_to_check = ['CS_Pseudomonas aeruginosa', 'P12M_2_Pseudomonas aeruginosa', 'N12M_Pseudomonas aeruginosa']\n",
    "colours = metadata[columns_to_check].isin([1]).any(axis=1).map({True: 'blue', False: 'red'})\n",
    "\n",
    "# Plot the PCA results\n",
    "plt.figure(figsize=(16, 12))\n",
    "plt.scatter(pca_df['PC1'], pca_df['PC2'], alpha=0.2, c=colours, s=sizes)\n",
    "plt.title(f'{genus} detected by {sequence_type} sequencing coloured whether positive - past, present, or future' + f\"\\nMedian number of '{genus}' sequences: {md:.0f}\")\n",
    "plt.xlabel(f'PC1 ({pc1_variance:.3f}%)')\n",
    "plt.ylabel(f'PC2 ({pc2_variance:.3f}%)')\n",
    "\n",
    "# add the loadings ... we only plot maxloadings here\n",
    "maxloadings = 15\n",
    "if len(loadings) < maxloadings:\n",
    "    maxloadings = len(loadings)\n",
    "\n",
    "plotscaler = 2\n",
    "texts = []\n",
    "colour_cycle = cycle(mcolors.TABLEAU_COLORS)\n",
    "found_pseudomonas = False\n",
    "for i in range(maxloadings):\n",
    "    c = next(colour_cycle)\n",
    "    if top_loadings_df.index[i] == 'Pseudomonas':\n",
    "        found_pseudomonas = True\n",
    "        c = 'black'\n",
    "    xpos = top_loadings_df.iloc[i, 0]*plotscaler\n",
    "    ypos = top_loadings_df.iloc[i, 1]*plotscaler\n",
    "    plt.arrow(0, 0, xpos, ypos, \n",
    "              color=c, alpha=0.5, width=0.05)\n",
    "    texts.append(plt.text(xpos, ypos, top_loadings_df.index[i], color=c))\n",
    "\n",
    "if not found_pseudomonas:\n",
    "    # specifically add Pseudomonas\n",
    "    pidx = top_loadings_df.index.get_loc('Pseudomonas')\n",
    "    xpos = top_loadings_df.iloc[pidx, 0]*plotscaler\n",
    "    ypos = top_loadings_df.iloc[pidx, 1]*plotscaler\n",
    "    plt.arrow(0, 0, xpos, ypos, color='black', alpha=0.5, width=0.05)\n",
    "    texts.append(plt.text(xpos, ypos, top_loadings_df.index[pidx], color='black'))\n",
    "\n",
    "    \n",
    "adjust_text(texts)\n",
    "\n",
    "# Add a legend\n",
    "blue_patch = plt.Line2D([0], [0], marker='o', color='w', label='Pseudomonas culture positive', \n",
    "                         markerfacecolor='blue', alpha=0.2, markersize=10)\n",
    "red_patch = plt.Line2D([0], [0], marker='o', color='w', label='Pseudomonas culture negative', \n",
    "                        markerfacecolor='red', alpha=0.2, markersize=10)\n",
    "small_patch = plt.scatter([], [], color='k', label=f'{bins[1]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[0])\n",
    "med1_patch = plt.scatter([], [], color='k', label=f'{bins[1]:.0f} - {bins[2]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[1])\n",
    "med2_patch = plt.scatter([], [], color='k', label=f'{bins[2]:.0f} - {bins[3]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[2])\n",
    "big_patch = plt.scatter([], [], color='k', label=f'>{bins[3]:.0f} sequences', \n",
    "                        alpha=0.2, s=sizelist[3])\n",
    "\n",
    "plt.legend(handles=[blue_patch, red_patch, small_patch, med1_patch, med2_patch, big_patch])\n",
    "\n",
    "\n",
    "plt.savefig('img/MGI_genus_pseudomonas_past_present_future.png')\n",
    "# Show the plot\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "48fc75f4-1301-489c-aeb2-d9e1f4a511d9",
   "metadata": {},
   "source": [
    "# One pwCF Outlier\n",
    "\n",
    "In the subsystems data, we found a single outlier pwCF who has a lot of flagellar subsystems (see MGI_Subsystems.ipynb)\n",
    "\n",
    "df[[x for x in df.columns if x.startswith('1128691_')]].sort_values(by='1128691_20171206_S', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "74fd8a5b-03b3-4c5c-9e48-a485b51823e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.options.display.float_format = '{:,.2f}'.format\n",
    "wants = [x for x in df.index if x.startswith('1128691_')]\n",
    "df.loc[wants,['Pseudomonas', 'Streptococcus']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "ee8a0cac-e568-4f3b-9e05-d64938ec7346",
   "metadata": {},
   "outputs": [],
   "source": [
    "intpwcf = df.loc[wants,:]\n",
    "intpwcf = intpwcf.loc[:,intpwcf.sum() > 2000]\n",
    "intpwcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "1925806e-88f6-4e29-9bba-d1b3844d1109",
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.boxplot(data=intpwcf.loc[:,intpwcf.sum().sort_values(ascending=False).index])\n",
    "plt.xticks(rotation=90)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "01625ccc-a488-4b6b-9700-c691562b88d6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50328f8d-e621-443b-8aa6-0c54a7f57f16",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "BioinfieConda",
   "language": "python",
   "name": "bioinfieconda"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
