{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "07906f24-78e2-4b70-8d42-d0027a91fd35",
   "metadata": {},
   "source": [
    "# Mutual information between Reads and MAGs\n",
    "\n",
    "This notebook measures the overlap between the reads and the MAGs.\n",
    "\n",
    "We want to know which MAGs (genomes) are most strongly associated with the latent biological gradients (COPFs) that emerged from the autoencoder trained on taxonomy + function data.\n",
    "\n",
    "We use mutual information because:\n",
    " - it captures any relationship (linear or non-linear),\n",
    " - it’s robust to scale and zero-inflation,\n",
    " - and it tells you how much knowing one variable reduces uncertainty about the other.\n",
    "\n",
    "We start by recreating the latent dimensions from our autoencoder, and then correlate the MAG abundance with those dimensions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "54914dd7-d97a-4a8b-8097-9fc39ebb7208",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-11-03 18:50:27.484016: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: SSE4.1 SSE4.2 AVX AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "Lets go GPU\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import sys\n",
    "import torch\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras import layers, models\n",
    "\n",
    "if not torch.cuda.is_available():\n",
    "    print(\"FATAL: Please run this notebook in an environment with CUDA available\", file=sys.stderr)\n",
    "else:\n",
    "    print(\"Lets go GPU\", file=sys.stderr)\n",
    "    \n",
    "from socket import gethostname\n",
    "\n",
    "hostname = gethostname()\n",
    "\n",
    "if hostname.startswith('hpc-node'):\n",
    "    IN_DEEPTHOUGHT = True\n",
    "    sys.path.append('..')\n",
    "else:\n",
    "    IN_DEEPTHOUGHT = False\n",
    "from cf_analysis_lib.load_libraries import *\n",
    "import cf_analysis_lib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9a3cb01c-8c02-431e-b9f7-dbfc22406a91",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using ../MGI/FunctionalAnalysis/subsystems/MGI_subsystems_norm_ss.tsv.gz for the subsystems\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train: (101, 1512)\n",
      "Test: (26, 1512)\n"
     ]
    }
   ],
   "source": [
    "sequence_type = \"MGI\"\n",
    "datadir = '..'\n",
    "sslevel = 'subsystems_norm_ss.tsv.gz'\n",
    "taxa = \"family\"\n",
    "\n",
    "df, metadata = cf_analysis_lib.read_the_data(sequence_type, datadir, sslevel='subsystems_norm_ss.tsv.gz', taxa=\"family\")\n",
    "\n",
    "training_data, validate_data = train_test_split(df, test_size=0.2, random_state=42)\n",
    "print(f\"Train: {training_data.shape}\")\n",
    "print(f\"Test: {validate_data.shape}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d9508c4a-e0f8-4430-b8c4-1085f00773cc",
   "metadata": {},
   "source": [
    "## Read the autoencoder model and recreate the latent dimensions\n",
    "\n",
    "This uses the test and training data from above"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ecdcb839-8071-4771-8c8c-260886af4317",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading the autoencoder from disk.\n",
      "2025-11-03 18:50:42.148615: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 343 MB memory:  -> device: 0, name: Tesla V100-PCIE-32GB, pci bus id: 0000:41:00.0, compute capability: 7.0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4/4 [==============================] - 2s 2ms/step\n",
      "1/1 [==============================] - 0s 64ms/step\n",
      "4/4 [==============================] - 0s 4ms/step\n"
     ]
    }
   ],
   "source": [
    "outdir = 'cluster_gbrfs'\n",
    "if not os.path.exists(outdir):\n",
    "    raise Exception(f\"FATAL: Path does not exist: {outdir}\")\n",
    "print(f\"Loading the autoencoder from disk.\", file=sys.stderr)\n",
    "autoencoder = models.load_model(os.path.join(outdir, \"autoencoder_model\"), compile=False)\n",
    "autoencoder.compile(optimizer='adam', loss='mse')\n",
    "encoder = models.load_model(os.path.join(outdir, \"encoder_model\"), compile=False)\n",
    "decoder = models.load_model(os.path.join(outdir, \"decoder_model\"), compile=False)\n",
    "encoder.compile()\n",
    "decoder.compile()\n",
    "train_latent = encoder.predict(training_data)\n",
    "val_latent = encoder.predict(validate_data)\n",
    "latent_dimensions = encoder.predict(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e8071cc9-4b96-4544-8f92-84da8a0aa397",
   "metadata": {},
   "source": [
    "## Recreate the correlation matrix.\n",
    "\n",
    "We need this to look at the correlations between MAGs and latent dimensions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4f4e3734-052f-4004-bd15-f9e7040c8163",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- COPF 31 : Length 40 ---\n",
      "['Sulfur transfer pathway CsdAEL', 'Sulfate assimilation related cluster', 'Stress proteins YciF, YciE', 'Two partner secretion pathway (TPS)', 'Type 4 conjugative transfer system, IncI1 type', 'Type III secretion systems', 'Siderophore Pyoverdine', 'Trans-envelope signaling system VreARI in Pseudomonas', 'L-2-amino-4-methoxy-trans-3-butenoic acid synthesis', 'Extracellular matrix proteins (PEL) involved in glucose-rich biofilm formation in Pseudomonas', 'Extracellular matrix proteins (PSL) involved in mannose-rich biofilm formation in Pseudomonas', 'IbrA and IbrB: co-activators of prophage gene expression', 'Rhamnolipids in Pseudomonas', 'Respiratory Systems - Shewanella (updated)', 'Phenazine biosynthesis', 'Periplasmic nitrate reductase EC 1.7.99.4', 'Outer membrane porins in Pseudomonas and Acinetobacter', 'Mycobacterial gene cluster associated with resistance against FAS-II antibiotics', 'Mycobacterial FadE proteins Acyl-CoA dehydrogenase', 'Mycobacterial FadD proteins (fatty acid CoA- and AMP- ligases)', 'MexXY System of Pseudomonas aeruginosa', 'Heme and heme d1 biosynthesis from siroheme', 'Malonate decarboxylase', 'Salmochelin-mediated Iron Acquisition', 'Potential DNA repair cluster', 'Type III secretion system', 'Cholera toxin', 'Beta-lactamases Ambler class D', 'Pseudomonadaceae', 'Cytochrome c oxidase EC 1.9.3.1', 'ABC-type iron transport system', 'Aestuariirhabdaceae', 'Geothermobacteraceae', 'CMP-pseudaminic acid synthesis', 'Biogenesis of cbb3-type cytochrome c oxidases', 'Adenylylsulfate reductase', 'Coenzyme PQQ synthesis', 'Marinobacteraceae', 'Colwelliaceae', 'Acyclic terpene utilization']\n",
      "\n",
      "--- COPF 32 : Length 43 ---\n",
      "['tRNA modification, 5-oxyacetyl uridine (cmo5U34)', 'Glutathione-dependent pathway of formaldehyde detoxification', 'Flagellum', 'Multidrug Resistance, Tripartite Systems Found in Gram Negative Bacteria TEMP', 'Efflux ABC transporter YadGH', 'General Secretion Pathway', 'Microbulbiferaceae', 'Taurine and alkanesulfonate utilization as sulfur sources', 'Copper transport and blue copper proteins', 'Bacterial pathways for dimethylsulfoniopropionate and acrylate catabolism', 'Spongiibacteraceae', 'Sarcosine oxidases, monomeric and heterotetrameric', 'EnvZ and OmpR regulon', 'Tricarboxylate transport system', 'Chemotaxis in Escherichia coli and Bacillus subtilis', 'Vanillate and syringate utilization', 'Carbon storage regulator', 'Fusaric acid resistance cluster', 'Tolerance to colicin E2 and CreBC signal transduction system', 'Tol-Pal Cell Envelope Complex', 'Phosphoenolpyruvate phosphomutase', 'Nitrosomonadaceae', 'Type VI secretion system', 'Maricaulaceae', 'D-arabinose utilization', 'HMG CoA Synthesis', 'Protection from Reactive Oxygen Species', 'Tripartite multidrug efflux systems (of RND type) MdtABC-TolC', 'Phosphonate metabolism', 'Putative TldE-TldD proteolytic complex', 'Glutathione: Non-redox reactions', 'Possible heterocyst differentiation related cluster', 'Putrescine utilization', 'Polymyxin resistance, lipid A modifications with phosphoethanolamine', 'Iron siderophore sensor & receptor system', 'Polymyxin resistance, lipid A modifications with 4-amino-4-deoxy-L-arabinose', 'Polyadenylation, bacterial', 'Quorum sensing regulation in Pseudomonas', 'Polymyxin resistance', 'Tripartite multidrug efflux systems (of MFS type) in Gram-negative bacteria', 'Type III secretion system orphans', 'Tripartite multidrug efflux systems (of MFS type) EmrKY-TolC and EmrAB-TolC', 'Dihydropyrimidine dehydrogenase']\n",
      "\n",
      "--- COPF 40 : Length 4 ---\n",
      "['Nucleoside ABC transporter cluster', 'Uridine to beta-Alanine and Thymine to 3-aminoisobutanoate module', 'Maintenance of outer membrane lipid asymmetry', 'Tripartite multidrug efflux systems (of RND type) in Pseudomonas']\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Interesting cluster is 31\n"
     ]
    }
   ],
   "source": [
    "num_features = df.shape[1]  # 1512\n",
    "num_latent_dims = latent_dimensions.shape[1]       # 50 - the number of output layers\n",
    "\n",
    "corr_matrix = pd.DataFrame(index=df.columns, columns=[f\"LD_{i}\" for i in range(num_latent_dims)])\n",
    "\n",
    "for i in range(num_features):\n",
    "    for j in range(num_latent_dims):\n",
    "        try:\n",
    "            corr = np.corrcoef(df.iloc[:, i], latent_dimensions[:, j])[0, 1]\n",
    "            corr_matrix.iloc[i, j] = corr\n",
    "        except:\n",
    "            print(f\"Error predicting {i} and {j}\", file=sys.stderr)\n",
    "            break\n",
    "\n",
    "corr_matrix = corr_matrix.astype(float)\n",
    "\n",
    "# Calculate linkage on the rows (i.e., the features)\n",
    "Z = linkage(corr_matrix, method='ward', metric='euclidean')\n",
    "\n",
    "k = 150  # more clusters, smaller groups!\n",
    "clusters = fcluster(Z, k, criterion='maxclust')  # Each feature gets a cluster ID [1..k]\n",
    "\n",
    "# create a df with the cluster assignments\n",
    "cluster_assignments = pd.DataFrame({\n",
    "    \"Feature\": corr_matrix.index,  # the feature names\n",
    "    \"COPF\": clusters\n",
    "})\n",
    "cluster_assignments.sort_values(\"COPF\", inplace=True)\n",
    "grouped = cluster_assignments.groupby(\"COPF\")\n",
    "by_cluster = {str(cluster_id): group[\"Feature\"].tolist() for cluster_id, group in grouped}\n",
    "\n",
    "# we print out the interesting clusters to make sure it is the same cluster numbers we've been using previously\n",
    "# cluster 31 should be our Pseudomonas friend.\n",
    "interesting_cluster = None\n",
    "interesting_cluster_count = 0\n",
    "grouped = cluster_assignments.groupby(\"COPF\")\n",
    "for cluster_id, group in grouped:\n",
    "    if group['Feature'].str.contains('Pseudomonas').any():\n",
    "        if group['Feature'].str.contains('Pseudomonas').sum() > interesting_cluster_count:\n",
    "            interesting_cluster_count = group['Feature'].str.contains('Pseudomonas').sum()\n",
    "            interesting_cluster = cluster_id\n",
    "        print(f\"--- COPF {cluster_id} : Length {group['Feature'].shape[0]} ---\")\n",
    "        print(group[\"Feature\"].tolist())\n",
    "        print()\n",
    "print(f\"Interesting cluster is {interesting_cluster}\", file=sys.stderr)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db687cc2-effa-4172-9f27-f077c46fb5f4",
   "metadata": {},
   "source": [
    "## Read the MAG data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8814b2ca-3330-4645-a8b4-d9e3fbcb845a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MAG</th>\n",
       "      <th>CheckM</th>\n",
       "      <th>UID</th>\n",
       "      <th># genomes</th>\n",
       "      <th># markers</th>\n",
       "      <th># marker sets</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5+</th>\n",
       "      <th>Completeness</th>\n",
       "      <th>Contamination</th>\n",
       "      <th>Strain heterogeneity</th>\n",
       "      <th>BV-BRC Mash</th>\n",
       "      <th>contigs</th>\n",
       "      <th>total size</th>\n",
       "      <th>BV-BRC Protein</th>\n",
       "      <th>GTDB</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Bin Id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>mag_12</td>\n",
       "      <td>o__Pseudomonadales</td>\n",
       "      <td>(UID4488)</td>\n",
       "      <td>185</td>\n",
       "      <td>813</td>\n",
       "      <td>308</td>\n",
       "      <td>4</td>\n",
       "      <td>800</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>99.27</td>\n",
       "      <td>1.6</td>\n",
       "      <td>72.73</td>\n",
       "      <td>Pseudomonas</td>\n",
       "      <td>406</td>\n",
       "      <td>6428243</td>\n",
       "      <td>Pseudomonas</td>\n",
       "      <td>Pseudomonas</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           MAG              CheckM        UID  # genomes  # markers  \\\n",
       "Bin Id                                                                \n",
       "12      mag_12  o__Pseudomonadales  (UID4488)        185        813   \n",
       "\n",
       "        # marker sets  0    1  2  3  4  5+  Completeness  Contamination  \\\n",
       "Bin Id                                                                    \n",
       "12                308  4  800  8  1  0   0         99.27            1.6   \n",
       "\n",
       "        Strain heterogeneity  BV-BRC Mash  contigs  total size BV-BRC Protein  \\\n",
       "Bin Id                                                                          \n",
       "12                     72.73  Pseudomonas      406     6428243    Pseudomonas   \n",
       "\n",
       "               GTDB  \n",
       "Bin Id               \n",
       "12      Pseudomonas  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mag_metadata = cf_analysis_lib.read_mag_metadata()\n",
    "# which MAG looks like Pseudomonas ... is it odd that we only have one?\n",
    "mag_metadata[(mag_metadata['BV-BRC Mash'].astype(str).str.contains('Pseudomo', case=False)) | (mag_metadata['CheckM'].astype(str).str.contains('Pseudomo', case=False))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e0a83200-b727-4f67-8754-5f6c658b4f6a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['MAG',\n",
       " 'CheckM',\n",
       " 'UID',\n",
       " '# genomes',\n",
       " '# markers',\n",
       " '# marker sets',\n",
       " '0',\n",
       " '1',\n",
       " '2',\n",
       " '3',\n",
       " '4',\n",
       " '5+',\n",
       " 'Completeness',\n",
       " 'Contamination',\n",
       " 'Strain heterogeneity',\n",
       " 'BV-BRC Mash',\n",
       " 'contigs',\n",
       " 'total size',\n",
       " 'BV-BRC Protein',\n",
       " 'GTDB']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(mag_metadata.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "b430162a-6522-4cad-8fa2-4c6602fb486f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MAG coverage has shape (127, 3215)\n",
      "Latent dimensions has shape (127, 50)\n"
     ]
    }
   ],
   "source": [
    "mag_coverage = cf_analysis_lib.read_mag_coverage()\n",
    "# For Mag Coverage (M): log-transform MAG abundances to stabilise variance.\n",
    "M_log = np.log10(mag_coverage + 1e-6)\n",
    "print(f\"MAG coverage has shape {M_log.shape}\")\n",
    "# For latent_dimensions (Z): scale each latent dimension to zero mean and unit variance\n",
    "Z_scaled = StandardScaler().fit_transform(latent_dimensions)\n",
    "print(f\"Latent dimensions has shape {Z_scaled.shape}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "acc43019-d71b-441c-bdd5-3183eca2fc1e",
   "metadata": {},
   "source": [
    "# Calculate a mutual information matrix\n",
    "\n",
    "mi_matrix[j, k] = how much information MAG j shares with latent dimension k."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4529c23c-1acc-483d-9009-9cf3419d89ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "mi_matrix = np.zeros((M_log.shape[1], Z_scaled.shape[1]))\n",
    "\n",
    "for k in range(Z_scaled.shape[1]):\n",
    "    mi_matrix[:, k] = mutual_info_regression(M_log, Z_scaled[:, k], random_state=42)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "088b912e-cca5-47b3-8eb4-7dfb6b4f12ca",
   "metadata": {},
   "source": [
    "We convert this to a dataframe and normalise it so that for every LD, the strongest MAG association = 1 and all others fall between 0 and 1."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "556a83ad-d5d0-43f0-931e-395d9b8524c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "mi_df = pd.DataFrame(mi_matrix, index=mag_coverage.columns, columns=[f\"LD_{i+1}\" for i in range(mi_matrix.shape[1])])\n",
    "mi_norm_df = mi_df.div(mi_df.max(axis=0), axis=1)\n",
    "mi_norm_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ba865460-6f52-4f5e-9bdf-211515b00bfc",
   "metadata": {},
   "source": [
    "## Identify the most informative MAGs for each latent dimension\n",
    "\n",
    "We keep every MAG whose MI ≥ 95th percentile for that latent dimension"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08f2e7eb-dc35-4539-9125-54acd0c588ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "top95 = {}\n",
    "for k, col in enumerate(mi_norm_df.columns):\n",
    "    cutoff = mi_norm_df[col].quantile(0.95)\n",
    "    top95[col] = mi_norm_df.index[mi_norm_df[col] >= cutoff].tolist()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a44c9551-ff7b-46fb-a1f4-f41a8bc3bcd7",
   "metadata": {},
   "outputs": [],
   "source": [
    "records = []\n",
    "for ld in mi_norm_df.columns:\n",
    "    for mag in mi_norm_df.index:\n",
    "        records.append({\n",
    "            \"MAG\": mag,\n",
    "            \"LatentDim\": ld,\n",
    "            \"MI_norm\": mi_norm_df.loc[mag, ld]\n",
    "        })\n",
    "\n",
    "mi_long = pd.DataFrame.from_records(records)\n",
    "mi_long[mi_long[\"MI_norm\"] > 0.95]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce311a74-df10-4319-8778-f0e329833f64",
   "metadata": {},
   "source": [
    "## Compare the relationship between individual MAGs and latent dimensions\n",
    "\n",
    "Which latent dimensions most closely match our _Pseudomonas_ MAG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d299371-9034-4345-a72c-1f32232ca106",
   "metadata": {},
   "outputs": [],
   "source": [
    "mi_long[mi_long[\"MAG\"]=='mag_12'][['LatentDim', 'MI_norm']].sort_values(by='MI_norm', ascending=False).head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e070e295-0a31-44ed-ba92-0c055d0156f4",
   "metadata": {},
   "source": [
    "Identify top latent dimensions for each MAG\n",
    "\n",
    "Each MAG will have high MI with a small subset of latent dimensions. \n",
    "\n",
    "This tells us for each MAG, which latent dimensions it’s most strongly associated with."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2fa4bddb-1c88-4ed6-9df0-e01f27647f8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "top_ld_per_mag = {}\n",
    "threshold = 0.95  # or any quantile cutoff you chose earlier\n",
    "\n",
    "for mag, row in mi_norm_df.iterrows():\n",
    "    top_ld = row[row >= threshold].index.tolist()\n",
    "    top_ld_per_mag[mag] = top_ld\n",
    "top_ld_per_mag['mag_12']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2b487837-1dc1-4340-b59b-40537dccbac5",
   "metadata": {},
   "outputs": [],
   "source": [
    "ld_per_mag = [len(top_ld_per_mag[x]) for x in top_ld_per_mag]\n",
    "g = sns.histplot(ld_per_mag)\n",
    "l = g.set_xlabel('Number of LDs per mag')\n",
    "l = g.set_ylabel('Number of MAGs')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae8709ce-50c1-4869-af8d-668cb3256f93",
   "metadata": {},
   "source": [
    "From the correlation matrix `corr_matrix` above, we already know which features (taxa + functions) correlate with each latent dimension.\n",
    "\n",
    "We can assign each latent dimension to clusters that contain its top-correlated features:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1377bb6a-db96-4adc-8c3d-34d60fedbe22",
   "metadata": {},
   "outputs": [],
   "source": [
    "ld_to_clusters = {}\n",
    "\n",
    "for ld in corr_matrix.columns:\n",
    "    # Find features most strongly correlated with this LD (top 10% absolute correlation)\n",
    "    threshold = corr_matrix[ld].abs().quantile(0.90)\n",
    "    top_features = corr_matrix.index[corr_matrix[ld].abs() >= threshold]\n",
    "    \n",
    "    # Find which clusters those features belong to\n",
    "    clusters_for_ld = cluster_assignments.loc[\n",
    "        cluster_assignments[\"Feature\"].isin(top_features), \"COPF\"\n",
    "    ].unique().tolist()\n",
    "    \n",
    "    ld_to_clusters[ld] = clusters_for_ld\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f8f978b-4103-42e4-aad4-882d6a61d1ae",
   "metadata": {},
   "source": [
    "## Combine mags and clusters\n",
    "\n",
    "Now we get to the crux of the story - which MAGs are correalted with which clusters (and vice-versa). We need to use the latent dimensions as our intermediary, but ultimately this is where the biology gets interesting"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10fda1bf-737b-4911-ac62-12bd77638c61",
   "metadata": {},
   "outputs": [],
   "source": [
    "mag_to_clusters = {}\n",
    "clusters_to_mags = defaultdict(list)\n",
    "\n",
    "for mag, lds in top_ld_per_mag.items():\n",
    "    clusters = set()\n",
    "    for ld in lds:\n",
    "        clusters.update(ld_to_clusters.get(ld, []))\n",
    "    mag_to_clusters[mag] = sorted(clusters)\n",
    "    for c in clusters:\n",
    "        clusters_to_mags[str(c)].append(mag)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9d0c5ba-1a0a-432d-b97d-151c6c8877e8",
   "metadata": {},
   "source": [
    "## Explore some interesting clusters or MAGs\n",
    "\n",
    "For our first test, we are curious which MAGs correlate with cluster 31 - the _Pseudomonas_ cluster. \n",
    "\n",
    "Note that the only MAG annotated as coming from _Pseudomonas_ is **mag_12**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "764aba8e-124f-499c-abf0-b69b862eb52a",
   "metadata": {},
   "outputs": [],
   "source": [
    "ic = clusters_to_mags['31']\n",
    "print(f\"There are {len(mag_metadata[mag_metadata['MAG'].isin(ic)].index)} MAGs correlated with 31\", file=sys.stderr)\n",
    "mag_metadata[mag_metadata['MAG'].isin(ic)].sort_values(by='Bin Id')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9918f2e8-8c09-4e32-ac2d-f1c4002c4db6",
   "metadata": {},
   "source": [
    "### Mag_12 doesn't correlate with any clusters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0dd660ad-190b-4e4b-a652-6ee6f24249a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "mag_to_clusters['mag_12']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8580d362-8b82-4f82-9ca7-d41e44d065e3",
   "metadata": {},
   "source": [
    "### These are some other COPFs we are interested in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1a344ed3-df39-45ee-b4f8-983ec9a94f29",
   "metadata": {},
   "outputs": [],
   "source": [
    "cpafs = [29, 68, 82]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8526ce94-ba25-45dd-92ed-e60d7a4784d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "clusters_to_mags['82']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a134cdc4-9c06-4e93-b9ef-596bfc421124",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ac0d0ae-c6a3-4a9c-9e6b-765fd00359ed",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a2e46a1d-b164-42ae-9bba-72fbf7fabb27",
   "metadata": {},
   "outputs": [],
   "source": [
    "mag_names = sorted(mag_to_clusters.keys())\n",
    "cluster_ids = sorted(cluster_assignments[\"COPF\"].unique())\n",
    "assoc = pd.DataFrame(0, index=mag_names, columns=cluster_ids)\n",
    "\n",
    "for mag, clusts in mag_to_clusters.items():\n",
    "    assoc.loc[mag, clusts] = 1\n",
    "\n",
    "# drop empty rows\n",
    "assoc = assoc[assoc.sum(axis=1) > 0]\n",
    "# drop empty columns\n",
    "# assoc = assoc.loc[:, assoc.sum(axis=0) > 0]\n",
    "\n",
    "fig, ax = plt.subplots(figsize=(18,12))\n",
    "sns.heatmap(assoc, cmap=\"Greens\", cbar=False, ax=ax)\n",
    "\n",
    "plt.xlabel(\"COPF\")\n",
    "plt.ylabel(\"MAGs\")\n",
    "ax.tick_params(axis='both', labelsize=6)\n",
    "ax.yaxis.label.set_fontsize(8)\n",
    "ax.xaxis.label.set_fontsize(8)\n",
    "plt.title(\"MAG–COPF Associations via Latent Dimensions\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2044cdd9-1d9f-4401-af9e-cb75e5709ee8",
   "metadata": {},
   "outputs": [],
   "source": [
    "mag_metadata[mag_metadata['MAG'].isin(assoc.index)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69deefbd-3320-4b1a-b14f-9afd4710aa40",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea682d20-bceb-4362-9f39-05ee4195864c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a41a7627-e5cc-4e54-8735-050da4437cbb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5a2f37f5-e378-46ad-a992-6534ec5e03f6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "271956a6-d620-4d8e-8ef2-e45730855e7f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "gpu_notebook",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
