{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d72555a8-b621-4a40-8fce-2c5ca9cba5d3",
   "metadata": {},
   "source": [
    "# Metadata summary table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "837c54ce-c916-4e85-a718-df0b4755ebc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "from socket import gethostname\n",
    "\n",
    "# this is for ffmpeg\n",
    "os.environ['PATH'] += os.pathsep + os.path.expanduser('~/miniconda3/envs/bioinformatics/bin')\n",
    "\n",
    "hostname = gethostname()\n",
    "\n",
    "if hostname.startswith('hpc-node'):\n",
    "    IN_DEEPTHOUGHT = True\n",
    "    sys.path.append('..')\n",
    "else:\n",
    "    IN_DEEPTHOUGHT = False\n",
    "from cf_analysis_lib.load_libraries import *\n",
    "import cf_analysis_lib\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b3e4b98b-e759-407f-88c8-ff2c53e5bcd4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using ../MGI/FunctionalAnalysis/subsystems/MGI_subsystems_norm_ss.tsv.gz for the subsystems\n"
     ]
    }
   ],
   "source": [
    "sequence_type = \"MGI\"\n",
    "datadir = '..'\n",
    "sslevel = 'subsystems_norm_ss.tsv.gz'\n",
    "taxa = \"family\"\n",
    "\n",
    "df, metadata = cf_analysis_lib.read_the_data(sequence_type, datadir, sslevel='subsystems_norm_ss.tsv.gz', taxa=\"family\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f833404c-edc9-4b82-a0ad-cdbdf1838df7",
   "metadata": {},
   "source": [
    "## Person-level summaries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "09d98e44-c683-4cdb-8f7e-644b074013a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "meta = metadata.copy()\n",
    "# Ensure dates are datetime\n",
    "meta['Sample date'] = pd.to_datetime(meta['Sample date'])\n",
    "# One row per person for person-level summaries\n",
    "persons = meta.sort_values('Sample date').groupby('pwCF_ID', as_index=False).first()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "478d8b63-4742-41e2-ae14-7198ba97ee92",
   "metadata": {},
   "source": [
    "## Cohort size, samples, study period"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1fd4f42a-99a5-4ebb-86ce-f28292782e61",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "528"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n_people = meta['pwCF_ID'].nunique()\n",
    "n_samples = len(meta)\n",
    "\n",
    "start_date = meta['Sample date'].min()\n",
    "end_date = meta['Sample date'].max()\n",
    "duration_days = (end_date - start_date).days  # 528 in your text\n",
    "duration_days"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c380c2bf-17f4-44c9-8643-af68082cf40b",
   "metadata": {},
   "source": [
    "## Sex distribution (per person)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "90469ea6-e918-437c-88fb-1277afcacb68",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 35 females and 29 males in the study\n"
     ]
    }
   ],
   "source": [
    "sex_counts = persons['Gender'].value_counts().sort_index()\n",
    "# e.g. Female/Male\n",
    "n_female = sex_counts.get('F', 0)\n",
    "n_male   = sex_counts.get('M', 0)\n",
    "print(f\"There are {n_female} females and {n_male} males in the study\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a77f8264-119b-41ca-aa3c-3ab3ec67998c",
   "metadata": {},
   "source": [
    "## Age at first sampling, and under-18 vs adults"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2a222459-c2a9-4eb8-add1-5a6855d227b7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "37"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "age_first = persons['Age']\n",
    "\n",
    "median_age = age_first.median()\n",
    "age_q1 = age_first.quantile(0.25)\n",
    "age_q3 = age_first.quantile(0.75)\n",
    "\n",
    "n_under18 = (age_first < 18).sum()\n",
    "n_adult   = (age_first >= 18).sum()\n",
    "\n",
    "age_group_counts = persons['Age groups'].value_counts()\n",
    "paed_adult_counts = persons['Paediatric vs Adult'].value_counts()\n",
    "n_under18"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "323bc7fa-9cda-4943-b204-a5c522420a5c",
   "metadata": {},
   "source": [
    "## Pancreatic insufficiency and CFRD / impaired glucose tolerance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4d3d9ca4-b47c-4aad-a3ea-191cf9f153bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Pancreatic insufficiency\n",
    "pi_counts = persons['Pancreatic insufficiency (PI)'].value_counts()\n",
    "# e.g. if \"Yes\" marks PI:\n",
    "n_pi = pi_counts.get('Yes', 0)\n",
    "\n",
    "# CFRD / IGT (impaired glucose tolerance)\n",
    "cfrd_counts = persons['Cystic Fibrosis related diabetes (CFRD)'].value_counts()\n",
    "\n",
    "# Example if the column uses labels like \"CFRD\", \"Impaired glucose tolerance\", \"Normal\"\n",
    "n_cfrd = cfrd_counts.get('Yes', 0)\n",
    "n_igt  = cfrd_counts.get('Impaired', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9cd32c23-f029-41ba-9eec-65737ca6a005",
   "metadata": {},
   "source": [
    "## Pseudomonas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2b9633a0-e18c-4d1c-b4fb-c5a57d8c903a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 28 Pseudomonas +ve samples and 36 Pseudomonas negative samples\n"
     ]
    }
   ],
   "source": [
    "meta['CS_Pseudo_num'] = meta['CS_Pseudomonas aeruginosa'].astype('Int64')\n",
    "pseudo_any = meta.groupby('pwCF_ID')['CS_Pseudo_num'].max()\n",
    "\n",
    "n_pseudo_positive = (pseudo_any == 1).sum()\n",
    "n_pseudo_negative = (pseudo_any == 0).sum()\n",
    "print(f\"There are {n_pseudo_positive} Pseudomonas +ve samples and {n_pseudo_negative} Pseudomonas negative samples\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "15d7d108-811e-4ae7-9e8b-ff738c556020",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "There are 1 people on ivacaftor and 63 people not on ivacaftor\n"
     ]
    }
   ],
   "source": [
    "meta['on_ivacaftor'] = meta['3 Ivacaftor (Kalydeco)'].astype('Int64')\n",
    "iva_any = meta.groupby('pwCF_ID')['on_ivacaftor'].max()\n",
    "n_on_iva = (iva_any == 1).sum()\n",
    "n_off_iva = (iva_any == 0).sum()\n",
    "print(f\"There are {n_on_iva} people on ivacaftor and {n_off_iva} people not on ivacaftor\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "29376352-8434-42e0-9cde-4b0fc50cf5d1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1862551_20180123_S'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta[meta['3 Ivacaftor (Kalydeco)']==1].index[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "df88862b-d7d7-487d-aee8-1cb735fe3a89",
   "metadata": {},
   "source": [
    "## IP vs OP"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "668c7642-76db-445a-bc22-62700e3a67c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "ip_counts = meta['IP vs OP'].value_counts()\n",
    "n_inpatient_samples  = ip_counts.get('IP', 0)\n",
    "n_outpatient_samples = ip_counts.get('OP', 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2b92b4e8-5eb3-4ccb-900f-0f0608275e12",
   "metadata": {},
   "source": [
    "## Longitudinal coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d4cf7934-9552-49d5-b6ab-a0b4a3a5b152",
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_counts_per_person = meta['pwCF_ID'].value_counts()\n",
    "n_longitudinal = (sample_counts_per_person >= 2).sum()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "842c31e8-ffe4-46f8-a411-c03222783b92",
   "metadata": {},
   "source": [
    "## Summary table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "19dec692-a43e-4034-a130-c168ca00a0c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def combine_genes(a, b):\n",
    "    return \" / \".join(sorted([str(a), str(b)]))\n",
    "\n",
    "persons['CF_genotype'] = persons.apply(\n",
    "    lambda row: combine_genes(row['CF gene 1'], row['CF gene 2']),\n",
    "    axis=1\n",
    ")\n",
    "genotype_counts = persons['CF_genotype'].value_counts()\n",
    "# Choose top genotypes (you can pick how many to show)\n",
    "top_genotypes = genotype_counts.head(5)\n",
    "\n",
    "# Make a formatted string\n",
    "genotype_summary_str = \"; \".join(\n",
    "    [f\"{gt}: {count} ({count/len(persons):.0%})\"\n",
    "     for gt, count in top_genotypes.items()]\n",
    ")\n",
    "\n",
    "# Optionally add \"other\"\n",
    "other = len(persons) - top_genotypes.sum()\n",
    "if other > 0:\n",
    "    genotype_summary_str += f\"; Others: {other}\"\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "fad6a874-4361-4612-9dfb-e0375fbb00e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_rows = [\n",
    "    (\"Number of participants\", n_people),\n",
    "    (\"Number of samples\", n_samples),\n",
    "    (\"Study period\",\n",
    "     f\"{start_date.date()} – {end_date.date()} ({duration_days} days)\"),\n",
    "    (\"Sex, female / male\",\n",
    "     f\"{n_female} ({n_female/n_people:.0%}) / {n_male} ({n_male/n_people:.0%})\"),\n",
    "    (\"Median age at first sampling (IQR)\",\n",
    "     f\"{median_age:.1f} years ({age_q1:.1f}–{age_q3:.1f})\"),\n",
    "    (\"Participants <18 years / ≥18 years\",\n",
    "     f\"{n_under18} ({n_under18/n_people:.0%}) / {n_adult} ({n_adult/n_people:.0%})\"),\n",
    "    (\"Pancreatic insufficiency\",\n",
    "     f\"{n_pi} ({n_pi/n_people:.0%})\"),\n",
    "    (\"CFRD\",\n",
    "     f\"{n_cfrd} ({n_cfrd/n_people:.0%})\"),\n",
    "    (\"Impaired glucose tolerance\",\n",
    "     f\"{n_igt} ({n_igt/n_people:.0%})\"),\n",
    "    (\"Pseudomonas infection\",\n",
    "     f\"{n_pseudo_positive} ({n_pseudo_positive/n_people:.0%})\"),\n",
    "    (\"Inpatient samples\",\n",
    "     f\"{n_inpatient_samples}\"),\n",
    "    (\"Participants with ≥2 samples\",\n",
    "     f\"{n_longitudinal}\"),\n",
    "    (\"CFTR genotype (top five)\",\n",
    "    genotype_summary_str)\n",
    "]\n",
    "\n",
    "summary_table = pd.DataFrame(summary_rows,\n",
    "                             columns=[\"Characteristic\", \"Value\"])\n",
    "summary_table.to_excel(\"summary_table.xlsx\", sheet_name=\"Metadata Summary\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "620be63f-86eb-486d-a1a9-fd9bf294d940",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Characteristic</th>\n",
       "      <th>Value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Number of participants</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Number of samples</td>\n",
       "      <td>127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Study period</td>\n",
       "      <td>2017-09-18 – 2019-02-28 (528 days)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Sex, female / male</td>\n",
       "      <td>35 (55%) / 29 (45%)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Median age at first sampling (IQR)</td>\n",
       "      <td>16.5 years (11.0–26.5)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Participants &lt;18 years / ≥18 years</td>\n",
       "      <td>37 (58%) / 27 (42%)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Pancreatic insufficiency</td>\n",
       "      <td>59 (92%)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CFRD</td>\n",
       "      <td>9 (14%)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Impaired glucose tolerance</td>\n",
       "      <td>7 (11%)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Pseudomonas infection</td>\n",
       "      <td>28 (44%)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Inpatient samples</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Participants with ≥2 samples</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CFTR genotype (top five)</td>\n",
       "      <td>F508 / F508: 41 (64%); ? / F508: 7 (11%); F508...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Characteristic  \\\n",
       "0               Number of participants   \n",
       "1                    Number of samples   \n",
       "2                         Study period   \n",
       "3                   Sex, female / male   \n",
       "4   Median age at first sampling (IQR)   \n",
       "5   Participants <18 years / ≥18 years   \n",
       "6             Pancreatic insufficiency   \n",
       "7                                 CFRD   \n",
       "8           Impaired glucose tolerance   \n",
       "9                Pseudomonas infection   \n",
       "10                   Inpatient samples   \n",
       "11        Participants with ≥2 samples   \n",
       "12            CFTR genotype (top five)   \n",
       "\n",
       "                                                Value  \n",
       "0                                                  64  \n",
       "1                                                 127  \n",
       "2                  2017-09-18 – 2019-02-28 (528 days)  \n",
       "3                                 35 (55%) / 29 (45%)  \n",
       "4                              16.5 years (11.0–26.5)  \n",
       "5                                 37 (58%) / 27 (42%)  \n",
       "6                                            59 (92%)  \n",
       "7                                             9 (14%)  \n",
       "8                                             7 (11%)  \n",
       "9                                            28 (44%)  \n",
       "10                                                 66  \n",
       "11                                                 29  \n",
       "12  F508 / F508: 41 (64%); ? / F508: 7 (11%); F508...  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "fc50b3c3-2f6b-4fac-b639-fafceb616c36",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'F508 / F508: 41 (64%); ? / F508: 7 (11%); F508 / G551D: 4 (6%); A455 / F508: 1 (2%); F508 / p.Arg851: 1 (2%); Others: 10'"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genotype_summary_str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb1359a8-117c-4cb0-ab1a-baf522a63f87",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "gpu_notebook",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
