# Bioproject PRJNA825831

This sample comes from the [NCBI BioProject PRJNA825831](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA825831).

# Title:

CF airway microbiome Raw sequence reads

# Description:


CF respiratory samples obtained at pulmonary exacerbation, end of antibiotic treatment, and follow up


# Metadata:
We have included the metadata in two separate files, a [JSON format file](PRJNA825831.metadata.json.gz) that is computer readable, or a [tsv format file](PRJNA825831.metadata.tsv.gz) that you can import into excel or similar.

# Publications


Hahn A, Burrell A, Chaney H, Sami I, Koumbourlis AC, Freishtat RJ, Crandall KA, Zemanick ET. 2023. Therapeutic beta-lactam dosages and broad-spectrum antibiotics are associated with reductions in microbial richness and diversity in persons with cystic fibrosis. [Sci Rep 13:1217](https://doi.org/10.1038/s41598-023-27628-x).
  

Shumyatsky G, Burrell A, Chaney H, Sami I, Koumbourlis AC, Freishtat RJ, Crandall KA, Zemanick ET, Hahn A. 2022. Using metabolic potential within the airway microbiome as predictors of clinical state in persons with cystic fibrosis. ]Front Med 9:1082125](https://doi.org/10.3389/fmed.2022.1082125).
  
# Analysis:

- The entire data set is 117 runs, and 58,764,823,681 bp (including amplicon sequences)
- We analysed 114 metagenomic sequence runs.
- We predicted 12 samples out of 114 (10.5%) have _Pseudomonas aeruginosa_


## Prediction outcomes

Sample | Pseudomonas Prediction | Confidence | Certainty
 --- | --- | --- | ---
SRR19051531 | Negative | 0.51 | Low
SRR19051532 | Positive | 0.65 | Medium
SRR19051533 | Positive | 0.66 | Medium
SRR19051534 | Negative | 0.65 | Medium
SRR19051535 | Negative | 0.51 | Low
SRR19051536 | Positive | 0.68 | Medium
SRR19051537 | Positive | 0.74 | Medium
SRR19051538 | Positive | 0.69 | Medium
SRR19051539 | Negative | 0.79 | Medium
SRR19051540 | Negative | 0.51 | Low
SRR19051541 | Positive | 0.59 | Low
SRR19051542 | Negative | 0.78 | Medium
SRR19051543 | Negative | 0.63 | Medium
SRR19051544 | Negative | 0.83 | High
SRR19051545 | Negative | 0.67 | Medium
SRR19051546 | Positive | 0.68 | Medium
SRR19051547 | Negative | 0.81 | High
SRR19051548 | Negative | 0.63 | Medium
SRR19051549 | Negative | 0.74 | Medium
SRR19051550 | Negative | 0.67 | Medium
SRR19051551 | Negative | 0.80 | High
SRR19051552 | Negative | 0.79 | Medium
SRR19051554 | Negative | 0.77 | Medium
SRR19051555 | Negative | 0.79 | Medium
SRR19051556 | Negative | 0.70 | Medium
SRR19051557 | Negative | 0.84 | High
SRR19051558 | Negative | 0.82 | High
SRR19051559 | Negative | 0.76 | Medium
SRR19051560 | Negative | 0.82 | High
SRR19051561 | Negative | 0.78 | Medium
SRR19051562 | Negative | 0.69 | Medium
SRR19051563 | Negative | 0.63 | Medium
SRR19051565 | Negative | 0.80 | High
SRR19051566 | Negative | 0.57 | Low
SRR19051567 | Negative | 0.85 | High
SRR19051568 | Negative | 0.76 | Medium
SRR19051569 | Negative | 0.68 | Medium
SRR19051570 | Negative | 0.62 | Medium
SRR19051571 | Negative | 0.84 | High
SRR19051572 | Negative | 0.68 | Medium
SRR19051573 | Negative | 0.63 | Medium
SRR19051574 | Negative | 0.73 | Medium
SRR19051576 | Negative | 0.85 | High
SRR19051577 | Negative | 0.81 | High
SRR19051578 | Negative | 0.82 | High
SRR19051579 | Negative | 0.85 | High
SRR19051580 | Negative | 0.83 | High
SRR19051581 | Negative | 0.85 | High
SRR19051582 | Negative | 0.75 | Medium
SRR19051583 | Negative | 0.82 | High
SRR19051584 | Negative | 0.84 | High
SRR19051585 | Negative | 0.86 | High
SRR19051586 | Negative | 0.56 | Low
SRR19051587 | Negative | 0.78 | Medium
SRR19051588 | Negative | 0.59 | Low
SRR19051589 | Negative | 0.83 | High
SRR19051590 | Negative | 0.78 | Medium
SRR19051591 | Negative | 0.51 | Low
SRR19051592 | Positive | 0.68 | Medium
SRR19051593 | Negative | 0.69 | Medium
SRR19051594 | Positive | 0.70 | Medium
SRR19051595 | Negative | 0.79 | Medium
SRR19051596 | Negative | 0.79 | Medium
SRR19051597 | Negative | 0.78 | Medium
SRR19051598 | Negative | 0.58 | Low
SRR19051599 | Negative | 0.74 | Medium
SRR19051600 | Negative | 0.78 | Medium
SRR19051601 | Negative | 0.70 | Medium
SRR19051602 | Negative | 0.70 | Medium
SRR19051603 | Positive | 0.72 | Medium
SRR19051604 | Negative | 0.63 | Medium
SRR19051605 | Negative | 0.74 | Medium
SRR19051606 | Negative | 0.78 | Medium
SRR19051607 | Negative | 0.77 | Medium
SRR19051608 | Negative | 0.80 | High
SRR19051609 | Negative | 0.73 | Medium
SRR19051610 | Negative | 0.79 | Medium
SRR19051611 | Negative | 0.80 | Medium
SRR19051612 | Negative | 0.60 | Low
SRR19051613 | Negative | 0.58 | Low
SRR19051614 | Negative | 0.77 | Medium
SRR19051615 | Negative | 0.78 | Medium
SRR19051616 | Negative | 0.76 | Medium
SRR19051617 | Negative | 0.89 | High
SRR19051618 | Negative | 0.86 | High
SRR19051619 | Positive | 0.65 | Medium
SRR19051620 | Negative | 0.53 | Low
SRR19051621 | Negative | 0.51 | Low
SRR19051622 | Negative | 0.67 | Medium
SRR19051623 | Negative | 0.67 | Medium
SRR19051624 | Negative | 0.72 | Medium
SRR19051625 | Negative | 0.60 | Low
SRR19051626 | Negative | 0.60 | Medium
SRR19051627 | Negative | 0.59 | Low
SRR19051628 | Negative | 0.64 | Medium
SRR19051629 | Negative | 0.79 | Medium
SRR19051630 | Negative | 0.83 | High
SRR19051631 | Negative | 0.78 | Medium
SRR19051632 | Negative | 0.60 | Low
SRR19051633 | Negative | 0.77 | Medium
SRR19051634 | Negative | 0.70 | Medium
SRR19051635 | Negative | 0.82 | High
SRR19051636 | Negative | 0.80 | High
SRR19051637 | Negative | 0.76 | Medium
SRR19051638 | Negative | 0.71 | Medium
SRR19051639 | Negative | 0.66 | Medium
SRR19051640 | Negative | 0.54 | Low
SRR19051641 | Positive | 0.54 | Low
SRR19051642 | Negative | 0.82 | High
SRR19051643 | Negative | 0.61 | Medium
SRR19051644 | Negative | 0.75 | Medium
SRR19051645 | Negative | 0.78 | Medium
SRR19051646 | Negative | 0.71 | Medium
SRR19051647 | Negative | 0.75 | Medium


## t-SNE
![Comparison of Adelaide and PRJNA825831 samples by t-SNE](img/PRJNA825831_Pseudomonas_tSNE.png 'Fig. t-SNE of all the analysed sequence data coloured by whether Pseudomonas is predicted')


## PCA
![This cluster of features are most strongly associated with the presence of Pseudomonas](img/PRJNA825831_Pseudomonas_PCA.png 'Fig. PCA of the cluster of features most strongly associated with Pseudomonas colonization in PRJNA825831')


